src/odemis/gui/cont/tabs.py

```diff
@@ -3138,14 +3138,23 @@ def __init__(self, name, button, panel, main_frame, main_data):
         else:
             self.panel.pnl_focus.Show(False)
 
-        # Add autofocus in case there is a spectrometer after the optical fiber,
-        # and the spectrometer
-        # Consider the focuser is after the fiber, if the fiber aligner affects it
-        if (main_data.fibaligner and main_data.focus and
-            main_data.focus.name in main_data.fibaligner.affects.value):
-            if ccd_focuser:
-                logging.warning("Focus seems to both affect the 'ccd' and be after "
-                                "the optical fiber ('fiber-aligner').")
+        # Add autofocus in case there is a focusable spectrometer after the optical fiber.
+        # Pick the focuser which affects at least one component common with the
+        # fiber aligner.
+        fib_focuser = None
+        if main_data.fibaligner:
+            aligner_affected = set(main_data.fibaligner.affects.value)
+            for focuser in (main_data.spec_ded_focus, main_data.focus):
+                if focuser is None:
+                    continue
+                if set(focuser.affects.value) & aligner_affected:
```

Comment:
It's not `and`, it's `&`. In python, they are different operator (which are pronounced the same in english). I think the official name for `&` is "bitwise and", to avoid the confusion. When used on sets (which is the type of both variables), it returns the intersection. So basically, it checks whether there is at least of the "aligner_affected" detector present in the "focuser affected" detectors.
Fair enough, we could have an extra line comment about this ;-)

Reply:
Ah true. Sorry I think I already asked this some time before but forgot about it. Thanks! Makes sense now :)